web/views.py

```
@@ -8839,3 +8842,289 @@ def leave_session_waiting_room(request, course_slug):
+        import datetime
+        if timezone.is_naive(scheduled_at_raw):
+            scheduled_at = timezone.make_aware(scheduled_at_raw)
+        else:
+            scheduled_at = scheduled_at_raw
+        if scheduled_at <= timezone.now():
+            messages.error(request, "Session must be scheduled in the future.")
+            return redirect("schedule_session", request_id=request_id)
+        try:
+            duration = int(request.POST.get("duration_minutes", 60))
+            if duration < 15:
+                raise ValueError
```

Ruff (0.15.6)

[warning] Abstract `raise` to an inner function

(TRY301)